DuckDuckGo/MainViewController.swift

```diff
@@ -721,4 +726,109 @@ extension MainViewController: Themable {
+            UIKeyCommand(input: "f", modifierFlags: [.alternate, .command], action: #selector(keyboardFind)),
+            UIKeyCommand(input: UIKeyCommand.inputRightArrow, modifierFlags: [.command], action: #selector(keyboardBrowserForward)),
+            UIKeyCommand(input: UIKeyCommand.inputLeftArrow, modifierFlags: [.command], action: #selector(keyboardBrowserBack)),
+            UIKeyCommand(input: UIKeyCommand.inputUpArrow, modifierFlags: [], action: #selector(keyboardArrowUp)),
+            UIKeyCommand(input: UIKeyCommand.inputDownArrow, modifierFlags: [], action: #selector(keyboardArrowDown)),
+            UIKeyCommand(input: UIKeyCommand.inputEscape, modifierFlags: [], action: #selector(keyboardEscape))
+        ]
+    }
+    
+    @objc func keyboardFind() {
+        if let controller = homeController {
+            controller.launchNewSearch()
+        } else {
+            omniBar.becomeFirstResponder()
+        }
+    }
+    
+    @objc func keyboardEscape() {
+        
+        if let controller = autocompleteController {
+            controller.keyboardEscape()
+            homeController?.collectionView.omniBarCancelPressed()
+        } else if let controller = homeController {
+            controller.omniBarCancelPressed()
+        } else {
+            omniBar.resignFirstResponder()
+        }
+    }
+    
+    @objc func keyboardArrowDown() {
+        if let controller = autocompleteController {
+            controller.keyboardMoveSelectionDown()
+        } else {
+            currentTab?.webView.becomeFirstResponder()
+        }
+    }
+    
+    @objc func keyboardArrowUp() {
+        if let controller = autocompleteController {
+            controller.keyboardMoveSelectionUp()
+        } else {
+            currentTab?.webView.becomeFirstResponder()
+        }
+    }
+
+    @objc func keyboardNewTab() {
+        if let tab = currentTab {
+            tabDidRequestNewTab(tab)
+        } else {
+            keyboardFind()
+        }
+    }
+    
+    @objc func keyboardCloseTab() {
+        guard let tab = currentTab else { return }
+        closeTab(tab.tabModel)
+        if tabManager.count == 0 {
+            launchNewSearch()
+        }
+    }
+    
+    @objc func keyboardNextTab() {
+        guard let tab = currentTab else { return }
+        guard let index = tabManager.model.indexOf(tab: tab.tabModel) else { return }
+        let targetTabIndex = index + 1 >= tabManager.model.count ? 0 : index + 1
+        select(tabAt: targetTabIndex)
+    }
+    
+    @objc func keyboardPreviousTab() {
+        guard let tab = currentTab else { return }
+        guard let index = tabManager.model.indexOf(tab: tab.tabModel) else { return }
+        let targetTabIndex = index - 1 < 0 ? tabManager.model.count - 1 : index - 1
+        select(tabAt: targetTabIndex)
+    }
+    
+    @objc func keyboardShowAllTabs() {
+        showTabSwitcher()
```

Comment:
If I open all tabs screen from the home screen then I can't seem to navigate the tabs using arrows (no 'current tab' maybe?)

Reply:
Sorted, thanks